src/hotspot/cpu/ppc/ppc.ad

```diff
@@ -11504,6 +11505,37 @@ instruct cmpL3_reg_reg_ExEx(iRegIdst dst, iRegLsrc src1, iRegLsrc src2) %{
   %}
 %}
 
+instruct cmovI_conIvalueMinus1_conIvalue0_conIvalue1_bc(iRegIdst dst, flagsRegSrc crx) %{
+  effect(DEF dst, USE_DEF crx);
+  predicate(false);
+  ins_cost(DEFAULT_COST*3);
+
+  format %{ "SETBC   R0, GT\n\t"
+            "SETNBC  $dst, LT\n\t"
+            "OR      $dst, $dst, R0" %}
+
+  size(12);
+  ins_encode %{
+    __ setbc(R0, (($crx$$reg << 2) | 1) /* greater than */);
+    __ setnbc($dst$$Register, ($crx$$reg << 2) /* less than */);
+    __ or_unchecked($dst$$Register, $dst$$Register, R0);
```

Comment:
In general, I think it'd be better to use orr which makes sure we never unintentionally emit an instruction which modifies smt priority "smt_prio_...". In this case dst != R0, so this doesn't happen.

Reply:
Fixed!